rye/src/cli/publish.rs

```diff
@@ -103,50 +102,35 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
 
     let username = match cmd.username {
         Some(username) => username,
-        None => credentials
-            .get(repository)
-            .and_then(|table| table.get("username"))
+        None => repo_credentials
+            .get("username")
             .map(|username| username.to_string())
             .map(escape_string)
             .unwrap_or("__token__".to_string()),
     };
 
-    let token = if let Some(token) = cmd.token {
-        let secret = Secret::new(token);
-        let maybe_encrypted = maybe_encrypt(&secret, cmd.yes)?;
-        let maybe_encoded = maybe_encode(&secret, &maybe_encrypted);
-        credentials[repository]["token"] = Item::Value(maybe_encoded.expose_secret().into());
-
-        secret
-    } else if let Some(token) = credentials
-        .get(repository)
-        .and_then(|table| table.get("token"))
+    let maybe_secret = if let Some(token) = cmd.token {
+        secret_from_token(token, cmd.yes, repo_credentials)?
```

Comment:
Made this helper to deduplicate the common logic between this cmdline case and the prompt case below